packages/client/hmi-client/src/components/workflow/ops/simulate-ensemble-ciemss/tera-simulate-ensemble-node-ciemss.vue

```diff
@@ -1,53 +1,82 @@
+		<Button v-if="areInputsFilled" label="Open" @click="emit('open-drilldown')" severity="secondary" outlined />
+	</main>
 </template>
 
 <script setup lang="ts">
 import _ from 'lodash';
 import { ref, computed, watch } from 'vue';
-import { getRunResultCiemss, pollAction, getSimulation } from '@/services/models/simulation-service';
+import { pollAction, getSimulation, getRunResultCSV, parsePyCiemssMap } from '@/services/models/simulation-service';
 import Button from 'primevue/button';
 import { WorkflowNode, WorkflowPortStatus } from '@/types/workflow';
-import { RunResults } from '@/types/SimulateConfig';
 import { Poller, PollerState } from '@/api/api';
-import TeraSimulateChart from '@/components/workflow/tera-simulate-chart.vue';
-import TeraOperatorPlaceholder from '@/components/operator/tera-operator-placeholder.vue';
-import TeraProgressSpinner from '@/components/widgets/tera-progress-spinner.vue';
 import { logger } from '@/utils/logger';
-import { chartActionsProxy, nodeOutputLabel } from '@/components/workflow/util';
+import { nodeOutputLabel } from '@/components/workflow/util';
 import { useProjects } from '@/composables/project';
 import { createDatasetFromSimulationResult } from '@/services/dataset';
+import { useCharts } from '@/composables/useCharts';
+import { ModelConfiguration } from '@/types/Types';
+import { DataArray } from '@/utils/stats';
+import { useChartSettings } from '@/composables/useChartSettings';
+import { getChartEnsembleMapping, usePreparedChartInputs } from './simulate-ensemble-util';
 import {
 	SimulateEnsembleCiemssOperation,
 	SimulateEnsembleCiemssOperationState
 } from './simulate-ensemble-ciemss-operation';
+import TeraNodePreview from '../tera-node-preview.vue';
 
 const props = defineProps<{
 	node: WorkflowNode<SimulateEnsembleCiemssOperationState>;
 }>();
 const emit = defineEmits(['append-output', 'update-state', 'open-drilldown', 'append-input-port']);
 
-// const runResults = ref<RunResults>({});
-const runResults = ref<{ [runId: string]: RunResults }>({});
+const runResults = ref<{ [runId: string]: DataArray }>({});
+const runResultsSummary = ref<{ [runId: string]: DataArray }>({});
+const pyciemssMap = ref<Record<string, string>>({});
 const inProgressForecastId = computed(() => props.node.state.inProgressForecastId);
 const selectedRunId = ref<string>('');
+const areInputsFilled = computed(() => props.node.inputs.filter((ele) => ele.value).length > 1);
+// -------------- Charts && chart settings ----------------
+const chartSize = ref<{ width: number; height: number }>({ width: 180, height: 120 });
+const allModelConfigurations = ref<ModelConfiguration[]>([]);
+const preparedChartInputs = usePreparedChartInputs(props, runResults, runResultsSummary, pyciemssMap);
+const stateToModelConfigMap = ref<{ [key: string]: string[] }>({});
+const selectedOutputMapping = computed(() => getChartEnsembleMapping(props.node, stateToModelConfigMap.value));
+
+const { useEnsembleVariableCharts } = useCharts(
```

Comment:
I don't see any charts in the ensemble workflow node after running ensemble-simulate a couple of times. Can you describe what is happening here?

I configured this setup in the drilldown:

<img width="1012" alt="image" src="https://github.com/user-attachments/assets/ccf4ed6f-0917-489d-bd40-6399d24b88d3" />

Reply:
There is an issue around updating the `hideInNode` value for some charts. This setting can be seen if you click on the settings cog beside `Sus` or `I` in your output settings.

I believe @jryu01 is looking into the issue around updating this 

By default this value is set to true to not flood the node with charts the user has not explicitly asked to be shown.

Reply:
`I think hideInNode not working properly in calibrate ensemble is a separate issue.` 
I have tracked this down and [corrected this here]([7e78f51](https://github.com/DARPA-ASKEM/terarium/pull/6612/commits/7e78f51a8699fdb8d19e72136cc01f614baa74ad)) 

`I'm not sure if tera-node-preview handles that.`
I will check and correct as needed thank you

Reply:
> I have tracked this down and corrected this [here](https://github.com/DARPA-ASKEM/terarium/pull/6612/commits/7e78f51a8699fdb8d19e72136cc01f614baa74ad)

Nice thanks!

Reply:
Should be able to correctly update the `hideInNode` value now.
It should correctly show chart in node (when this setting is false)
<img width="570" alt="image" src="https://github.com/user-attachments/assets/e1cf1dfb-1e4d-4fff-83ba-727dfa2bf1e3" />